src/server/common/model/application/application.test.js

```diff
@@ -0,0 +1,43 @@
+import { Application } from './application.js'
+import { OnOffFarm } from '../answer/on-off-farm.js'
+
+const originDefaultState = {
+  onOffFarm: new OnOffFarm({ onOffFarm: 'on' }).toState(),
+  cphNumber: '12/123/1234',
+  address: {
+    addressLine1: 'Starfleet Headquarters',
+    addressLine2: '24-593 Federation Drive',
+    addressTown: 'San Francisco',
+    addressCounty: 'San Francisco',
+    addressPostcode: 'RG24 8RR'
+  }
+}
+
+const licenseDefaultState = {
+  emailAddress: 'name@example.com'
+}
+
+describe('Application', () => {
+  it('should create an Application instance from a valid state', () => {
+    const state = {
+      origin: originDefaultState,
+      license: licenseDefaultState
+    }
+
+    const application = Application.fromState(state)
+
+    expect(application).toBeInstanceOf(Application)
```

Comment:
I would need to review these tests as it doesn't look great that I have to check _data.origin._data...

Reply:
agreed that this is breaking encapsulation a little too much - but I definitely don't think that's a blocker for now.